packages/swarm-docs/src/components/nav.js

```diff
@@ -3,21 +3,45 @@ import * as React from 'react';
 import { Link } from 'gatsby';
 
 type Props = {
-    fileNames: Array<string>
+    files: Array<Object>
+};
+
+const transformFiles = (files: Array<Object>) => {
```

Comment:
It's difficult to follow what 'transformation' this function is making - it would be helpful to specify the output type. I think it's something like

```jsx
type FileDirectoryMaps = Array<{ [folderName: string]: Array<string> }>
```